app/views/common/os_selection/_pxe_loader.html.erb

```diff
@@ -0,0 +1,12 @@
+<%= fields_for item do |f| -%>
+  <% if @operatingsystem.available_loaders(item.architecture).any? -%>
+    <% selected_attr = f.object.new_record? && item.hostgroup.present? && item.hostgroup.pxe_loader.nil? ? { :selected => @operatingsystem.preferred_loader(item.architecture) } : {} -%>
```

Comment:
Oh, and this should be suggested even when there is no hostgroup. I know I gave the if statement to you, sorry. Can we move this to the model or a helper method and add some tests? This jumps back to the suggested value when I change the option, save the host and it comes back with validation errors.